packages/core/admin/ee/server/services/review-workflows/assigned-content-types.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const { mapAsync } = require('@strapi/utils');
+const { difference } = require('lodash/fp');
+const { getService } = require('../../utils');
+const { WORKFLOW_MODEL_UID } = require('../../constants/workflows');
+
+module.exports = () => ({
+  async create({ assigned, defaultStage }) {
+    await mapAsync(assigned, async (uid) => {
+      const swappingFromWorkflow = await this.getAssignedWorkflow(uid);
+      // Content type is already assigned to a workflow, swap it
+      if (swappingFromWorkflow) {
+        return this.swapAssignedContentType(swappingFromWorkflow, defaultStage, uid);
+      }
+      // Content type is not assigned to a workflow, init it
+      return this.initEntityStages(uid, defaultStage);
+    });
+  },
+
+  async update({ srcAssigned, destAssigned, defaultStage }) {
+    // Find difference between old assignedContentTypes and new ones, using lodash fp
+    const { created, deleted } = diffAssignedContentTypes(srcAssigned, destAssigned);
+
+    await mapAsync(created, async (uid) => {
+      const swappingFromWorkflow = await this.getAssignedWorkflow(uid);
+      // Content type is already assigned to a workflow, swap it
+      if (swappingFromWorkflow) {
+        return this.swapAssignedContentType(swappingFromWorkflow, defaultStage, uid);
+      }
+      // Content type is not assigned to a workflow, init it
+      return this.initEntityStages(uid, defaultStage);
+    });
+
+    await mapAsync(deleted, async (uid) => {
+      await getService('stages').deleteAllEntitiesStage(uid, {});
+    });
+  },
+
+  /**
+   * Swapping involves:
+   *  - Updating entity stages of the content type
+   *  - Updating the contentTypes of the previous workflow
+   * @param {Workflow} srcWorkflow
+   * @param {Workflow.Stage} stage
+   * @param {string} uid
+   */
+  async swapAssignedContentType(srcWorkflow, stage, uid) {
+    // Update entity stages of the content type
+    await getService('stages').updateAllEntitiesStage(uid, { toStageId: stage.id });
```

Comment:
I'm not sure but should we handle this in a transaction? I'm thinking that we need to ensure a content type is only in one list of assigned content types and an error here could cause it to end up in multiple?

Reply:
This should be handled by the transaction here https://github.com/strapi/strapi/pull/16722/files#diff-dfcb527cbe969391777b1f710fa4f7a410d38a2acc7ce093d7aedb39a62b5f41R22